dss/events/handlers/validator.py

```diff
@@ -0,0 +1,80 @@
+"""Utilities in this file are use to removed extra fields from index data for adding to the index."""
+import jsonschema
+import re
+from jsonschema.validators import create, RefResolver
+from jsonschema import _utils, _validators, validators
+
+import json
+from urllib.request import urlopen
+from urllib.error import HTTPError
+import ssl
+from typing import List
+import logging
+
+
+def read_json_url(link):
+    gcontext = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+    f = urlopen(link, context=gcontext)
+    return json.load(f)
+
+
+def extract_json_fields(json_data: dict, path: List[str], fields: List[str]):
+    """
+        json_data: The json data from which to remove the field
+        path: The path through the json data to find the field.
+        fields: A list of fields names to remove from the json data at the locations specified by path."""
+    current = json_data
+    if path:
+        for next in path:
+            current = current[next]
+    for field in fields:
+        current.pop(field)
+
+# def additionalProperties(validator, aP, instance, schema):
+#     """Removed any field no specified in the schema. aP is required to be a `validator` but is intentionally unused in
+#     this case."""
+#     if not validator.is_type(instance, "object"):
+#         return
+#     extras = set(_utils.find_additional_properties(instance, schema))
+#     for field in extras:
+#         instance.pop(field)
+#         logging.debug(f"removed field: '{field}'")
+
+DSS_Draft4Validator = create(
+    meta_schema=_utils.load_schema("draft4"),
+    validators={
+        u"$ref": _validators.ref,
+        u"additionalProperties": _validators.additionalProperties,
+        u"properties": _validators.properties_draft4,
+        u"required": _validators.required_draft4,
+    },
+    version="draft4",
+)
+
+resolver = RefResolver(referrer='', base_uri='')
+
+def scrub_index_data(index_data: dict) -> list:
+    extra_fields = []
+    for file in index_data.keys():
+        core = index_data[file].get('core', None)
+        if core is not None:
+            try:
+                schema = read_json_url(core['schema_url'])
+            except HTTPError:
```

Comment:
In this case, when the schema file could not be accessed, a warning should be logged that includes the schema url.